.github/workflows/build-and-upload.yaml

```diff
@@ -0,0 +1,205 @@
+name: Build Harvester Dashboard and Upload
+
+on:
+  workflow_call:
+    inputs:
+      CI_BRANCH:
+        required: false
+        type: string
+      CI_BUILD_TAG:
+        required: false
+        type: string
+
+env:
+  GOOGLE_AUTH: ''
+  DOCKER_USERNAME: ''
+  DOCKER_PASSWORD: ''
+  CI_BUILD_TAG: ${{inputs.CI_BUILD_TAG}}
+  CI_BRANCH: ${{inputs.CI_BRANCH}}
+  GIT_REPO: ${{github.repository}}
+  GIT_COMMIT: ${{github.sha}}
+  REPO: ${{github.event.repository.name || ''}}
+
+jobs:
+  # build-and-upload-docker-image:
+  #   name: Build & Upload Docker
+  #   runs-on: ubuntu-latest
+  #   permissions:
+  #     contents: read
+  #     packages: write
+  #     id-token: write
+  #   steps:
+  #     - uses: actions/checkout@v4
+  #       with:
+  #         fetch-depth: 1
+
+  #     - id: upload-gate
+  #       name: Upload Gate
+  #       run: ./scripts/build-upload-gate
+
+  #     - id: read-docker-secrets
+  #       name: Read Docker Secrets
+  #       uses: rancher-eio/read-vault-secrets@main
+  #       with:
+  #         secrets: |
+  #           secret/data/github/repo/${{ github.repository }}/dockerhub/harvester/credentials username | DOCKER_USERNAME ;
+  #           secret/data/github/repo/${{ github.repository }}/dockerhub/harvester/credentials password | DOCKER_PASSWORD ;
+
+  #     - id: login-docker
+  #       name: Docker Log in
+  #       uses: docker/login-action@v3
+  #       with:
+  #         username: ${{ env.DOCKER_USERNAME }}
+  #         password: ${{ env.DOCKER_PASSWORD }}
+
+  #     - id: push-docker
+  #       name: Docker Build
+  #       uses: docker/build-push-action@v5
+  #       with:
+  #         provenance: false
+  #         context: .
+  #         push: true
+  #         tags: rancher/harvester-ui:${{ github.ref_name }}
+
+  build-and-upload-hosted:
+    name: Build & Upload Hosted
```

Comment:
Not sure we still needed build host, from harvester/dashboard [README.md](https://github.com/harvester/dashboard#other-building-modes), this step is build `ui-index` file and push to GCP. e.g.
```
https://releases.rancher.com/harvester-ui/dashboard/latest/index.html
```